pacemaker_metrics.go

```diff
@@ -61,11 +66,136 @@ type resource struct {
 	NodesRunningOn int    `xml:"nodes_running_on,attr"`
 }
 
+// ***
+
+var (
+	pacemakerMetrics = metricDescriptors{
+		// the map key will function as an identifier of the metric throughout the rest of the code;
+		// it is arbitrary, but by convention we use the actual metric name
+		"nodes":           NewMetricDesc("pacemaker", "nodes", "The nodes in the cluster; one line per name, per status", []string{"name", "type", "status"}),
+		"nodes_total":     NewMetricDesc("pacemaker", "nodes_total", "Total number of nodes in the cluster", nil),
+		"resources":       NewMetricDesc("pacemaker", "resources", "The resources in the cluster; one line per id, per status", []string{"node", "id", "role", "managed", "status"}),
+		"resources_total": NewMetricDesc("pacemaker", "resources_total", "Total number of resources in the cluster", nil),
+	}
+
+	crmMonPath = "/usr/sbin/crm_mon"
+)
+
+func NewPacemakerCollector() (*pacemakerCollector, error) {
```

Comment:
regarding the Uppercase exported public functions, in theory since we are using only one package they aren't needed to be exported. Imho we should perhaps don't `Foo` but `foo` since we don't plan to expose them as public API really but I'm ok also like this.

Reply:
actually I guess in the future we'll likely move the various subsystems into separate packages, because we already have some name clashes with the XML parsing structs

Reply:
Ok let see later. was a historic remark for moment :grin: